pkg/sysinfo/sysinfo_linux_test.go

```diff
@@ -30,84 +27,43 @@ func TestReadProcBool(t *testing.T) {
 		t.Fatal("expected proc bool to be false, got true")
 	}
 
-	if readProcBool(path.Join(tmpDir, "no-exist")) {
+	if readProcBool(filepath.Join(tmpDir, "no-exist")) {
 		t.Fatal("should be false for non-existent entry")
 	}
 }
 
 func TestCgroupEnabled(t *testing.T) {
-	cgroupDir, err := os.MkdirTemp("", "cgroup-test")
-	assert.NilError(t, err)
-	defer os.RemoveAll(cgroupDir)
+	cgroupDir := t.TempDir()
 
 	if cgroupEnabled(cgroupDir, "test") {
 		t.Fatal("cgroupEnabled should be false")
 	}
 
-	err = os.WriteFile(path.Join(cgroupDir, "test"), []byte{}, 0o644)
-	assert.NilError(t, err)
+	err := os.WriteFile(filepath.Join(cgroupDir, "test"), []byte{}, 0o644)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	if !cgroupEnabled(cgroupDir, "test") {
 		t.Fatal("cgroupEnabled should be true")
 	}
 }
 
+// TestNew verifies that sysInfo is initialized with the expected values.
 func TestNew(t *testing.T) {
 	sysInfo := New()
-	assert.Assert(t, sysInfo != nil)
-	checkSysInfo(t, sysInfo)
-}
-
-func checkSysInfo(t *testing.T, sysInfo *SysInfo) {
-	// Check if Seccomp is supported, via CONFIG_SECCOMP.then sysInfo.Seccomp must be TRUE , else FALSE
-	if err := unix.Prctl(unix.PR_GET_SECCOMP, 0, 0, 0, 0); err != unix.EINVAL {
-		// Make sure the kernel has CONFIG_SECCOMP_FILTER.
-		if err := unix.Prctl(unix.PR_SET_SECCOMP, unix.SECCOMP_MODE_FILTER, 0, 0, 0); err != unix.EINVAL {
-			assert.Assert(t, sysInfo.Seccomp)
-		}
-	} else {
-		assert.Assert(t, !sysInfo.Seccomp)
+	if sysInfo == nil {
+		t.Fatal("sysInfo should not be nil")
 	}
-}
-
-func TestNewAppArmorEnabled(t *testing.T) {
-	// Check if AppArmor is supported. then it must be TRUE , else FALSE
-	if _, err := os.Stat("/sys/kernel/security/apparmor"); err != nil {
-		t.Skip("AppArmor Must be Enabled")
+	if expected := seccomp.IsEnabled(); sysInfo.Seccomp != expected {
```

Comment:
Starting to consider if we should create a local copy  of this instead  of  depending on  containerd's seccomp package, because it's really minimal. We could even consider moving it into our `profiles/seccomp` package, but perhaps that's not needed  if we only use it here.


(this is the only package where we import containerd's pkg/seccomp)